src/ensembl/src/content/app/entity-viewer/gene-view/components/transcripts-filter/TranscriptsFilter.test.tsx

```diff
@@ -0,0 +1,158 @@
+import { createTranscript } from 'tests/fixtures/entity-viewer/transcript';
+import TranscriptsFilter from './TranscriptsFilter';
+
+import RadioGroup from 'src/shared/components/radio-group/RadioGroup';
+
+const mockState = {
+  entityViewer: {
+    general: {
+      activeGenomeId: 'human',
+      activeEnsObjectIds: {
+        human: 'gene:brca2'
+      }
+    },
+    geneView: {
+      transcripts: {
+        human: {
+          'gene:brca2': {
+            expandedIds: [],
+            expandedDownloadIds: [],
+            filters: [],
+            sortingRule: 'default'
+          }
+        }
+      }
+    },
+    sidebar: {
+      human: {
+        status: Status.OPEN
+      }
+    }
+  }
+};
+
+const mockStore = configureMockStore([thunk]);
+
+const createProteinCodingTranscript = () => createTranscript();
+const createNonCodingTranscript = () => {
+  const transcript = createTranscript();
+  transcript.product_generating_contexts = [];
+  return transcript;
+};
+
+const proteinCodingTranscript1 = createProteinCodingTranscript();
+const proteinCodingTranscript2 = createProteinCodingTranscript();
+const nonCodingTranscript1 = createNonCodingTranscript();
+const nonCodingTranscript2 = createNonCodingTranscript();
+
+const defaultTranscripts = [
+  proteinCodingTranscript1,
+  proteinCodingTranscript2,
+  nonCodingTranscript1,
+  nonCodingTranscript2
+];
+
+const mockToggleFilter = jest.fn();
+
+const wrapInRedux = (
+  state: typeof mockState = mockState,
+  transcripts = defaultTranscripts
+) => {
+  return mount(
+    <Provider store={mockStore(state)}>
+      <TranscriptsFilter
+        transcripts={transcripts}
+        toggleFilter={mockToggleFilter}
+      />
+    </Provider>
+  );
+};
+
+describe('<TranscriptsFilter />', () => {
+  it('shows correct sorting option as selected', () => {
+    // default sorting option
+    let wrapper = wrapInRedux();
+    const defaultSortingLabel = wrapper
+      .find('label')
+      .findWhere((el) => el.text() === 'Default');
+    const defaultSortingRadioButton = defaultSortingLabel.find('input');
+    expect(defaultSortingRadioButton.prop('checked')).toBe(true);
+
+    // after we change sorting option
+    const updatedState = set(
+      'entityViewer.geneView.transcripts.human.gene:brca2.sortingRule',
+      'spliced_length_longest_to_shortest',
+      mockState
+    );
+    wrapper = wrapInRedux(updatedState);
+
+    const secondSortingLabel = wrapper
+      .find('label')
+      .findWhere(
+        (el) => el.text() === 'Combined exon length: longest – shortest'
+      );
+    const secondSortingRadioButton = secondSortingLabel.find('input');
+
+    expect(secondSortingRadioButton.prop('checked')).toBe(true);
+  });
+
+  it.skip('shows sort by protein for protein-coding transcripts', () => {
+    //todo once we get confirmation
+  });
+
+  it.skip('does not show sort by protein for non-coding transcripts', () => {
+    //todo once we get confirmation
+  });
```

Comment:
Since we aren't using protein-related filters in the near future, let's delete these two placeholder tests.